operators/src/processing/raster_type_conversion.rs

```diff
@@ -0,0 +1,233 @@
+use async_trait::async_trait;
+use futures::{stream::BoxStream, StreamExt, TryFutureExt, TryStreamExt};
+use geoengine_datatypes::{
+    primitives::{RasterQueryRectangle, SpatialPartition2D},
+    raster::{ConvertDataType, Pixel, RasterDataType, RasterTile2D},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::engine::{
+    ExecutionContext, InitializedRasterOperator, Operator, QueryContext, QueryProcessor,
+    RasterOperator, RasterQueryProcessor, RasterResultDescriptor, SingleRasterSource,
+    TypedRasterQueryProcessor,
+};
+use crate::util::Result;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct RasterTypeConversionParams {
+    output_data_type: RasterDataType,
+}
+
+pub type RasterTypeConversion = Operator<RasterTypeConversionParams, SingleRasterSource>;
```

Comment:
Maybe you can add some brief documentation on what this operator does.

Reply:
well it converts the rasters data type...

Reply:
[32b4043](https://github.com/geo-engine/geoengine/pull/606/commits/32b4043fff7f55daab305d3ccfc9f332fd38908e)